src/interface/src/app/account/delete-account-dialog/delete-account-dialog.component.ts

```diff
@@ -35,20 +35,28 @@ export class DeleteAccountDialogComponent {
   deleteAccount(): void {
     this.disableDeleteButton = true;
     this.authService
-      .deleteUser(
+      .deactivateUser(
         this.data.user,
         this.deleteAccountForm.get('currentPassword')?.value
       )
-      .subscribe(
-        (_) => {
+      .subscribe({
+        next: () => {
           this.dialogRef.close({
             deletedAccount: true,
           });
         },
-        (err) => {
-          this.error = err.error;
-          this.disableDeleteButton = false;
-        }
-      );
+        error: (err) => {
+          if (err.status === 403) {
+            this.error = 'Password was incorrect.';
+            this.disableDeleteButton = false;
+          } else if (err.status === 401) {
+            this.error = 'User is not logged in.';
+            this.disableDeleteButton = false;
```

Comment:
you can simplify this and put this outside your `if` condition instead of repeating it on each

Reply:
good catch- of course, just started being blind to it after a few edits